### ISSUE TITLE
nn.Dense bias_init is poorly documented and can

### ISSUE THREAD

**Issue by billmark** [google/flax#1386]
Provide as much information as possible. At least, this should include a description of your issue and steps to reproduce the problem. If possible also provide a summary of what steps or workarounds you have already tried.

### Steps to reproduce:
Whenever possible, please provide a *minimal example*. Please consider submitting it as a Colab link.

**Comment by marcvanzee** [comment]
Hi @billmark , thanks for your issue! Could you tell me in a bit more detail what you were missing from the description of `kernel_init` and `bias_init` arguments?

**Comment by billmark** [comment]
As I said in the original bug, "In particular, it's not clear what the parameters to the callback need to be."..  That is, what are the args passed to the kernel_init and bias_init callbacks?  How many of these args?  What types/shapes?  What do they mean?  etc.  There is *no* such information in the documentation.

**Comment by FelipeMartins96** [comment]
I found myself having the same problem reported when trying to reproduce the DDPG algorithm and an error was raised when trying to use the `lecun_uniform()` on bias initialization of `nn.Dense` params.

Is there a reason why the on the kernel initialization it sends inputs and outputs `(inputs.shape[-1], self.features)` and on bias only inputs `(self.features,)`?
https://github.com/google/flax/blob/c63d9c69205629fca00ad82135213151d656ce49/flax/linen/linear.py#L171-L173
vs
https://github.com/google/flax/blob/c63d9c69205629fca00ad82135213151d656ce49/flax/linen/linear.py#L179

From what I understood, this argument is passed on to the initializers as shape parameter.

About documentation: it seems to be missing on the docs a page about the initializers available at linen.initializers.
And for the [linen.Dense docs](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Dense.html#flax.linen.Dense.kernel_init) I think it could be included at the kernel init, and bias init documentation that the function will receive the arguments `key, shape, dtype`, where shape is  `(inputs.shape[-1], self.features)` for kernel and  `(self.features,)` for bias

**Comment by FelipeMartins96** [comment]
Had the same issue. I found this issue on jax repo which I think is relevant, the initializer functions are intentionally made to not work for 1D shapes. 
https://github.com/google/jax/issues/2075#issuecomment-578465814

I think maybe a page on the docs explaining the initializers would be interesting, or maybe link to the jax docs which lists the available functions. An example notebook explaining the module weights and biases initializations could be useful.

**Comment by marcvanzee** [comment]
Thanks for the feedback, I've updated the issue description. Indeed, we don't seem to have our initializers on RTD, while we do have the activation functions. We can simply add them here as well: https://github.com/google/flax/blob/main/docs/flax.linen.rst

**Comment by conorheins** [comment]
I've been having this same issue as well; namely I'm not sure what the argument to `kernel_init` should be. Shouldn't I be able to do this when creating a `nn.Dense()` layer, for instance?

```
layer = nn.Dense(features=16, kernel_init = nn.initializers.lecun_normal(scale = 0.1))
```

This throws the following error when doing `model.apply(params, input)`:

```
jax._src.traceback_util.UnfilteredStackTrace: TypeError: variance_scaling() got multiple values for argument 'scale'
```

It only works if I either A) don't provide an input argument to `nn.initializers.lecun_normal()` or B) use the following code instead, where I explicitly do the same thing as lecun_normal using the `variance_scaling` function.

```
nn.initializers.variance_scaling(0.1, "fan_in", "truncated_normal")
```

Sorry if this is not appropriate to paste here -- I can start a different issue if necessary. But it seems related to the problem of not knowing what `kernel_init` argument expects.

**Comment by jheek** [comment]
>>I t only works if I either A) don't provide an input argument to nn.initializers.lecun_normal() or B) use the following code instead, where I explicitly do the same thing as lecun_normal using the variance_scaling function.

I agree it's a bit frustating that you can only rescale the variance_scaling and not lecun_normal, kaiming, etc. These initializers come from the `jax.nn.initializers` module though so it's best to open an issue in google/jax about this

**Comment by cgarciae** [comment]
I'll look into this. I'll document the signature for each and clarify that `bias_initializer`s don't work with the `variance_scaling` family.

**Comment by marcvanzee** [comment]
@cgarciae any update on this?

**Comment by cgarciae** [comment]
@zaxtax will be taking the issue :)
